api-tests/src/test/java/org/openmrs/module/reporting/cohort/definition/evaluator/DrugOrderCohortDefinitionEvaluatorTest.java

```diff
@@ -0,0 +1,185 @@
+
+  }
+  
+  @Test
+  public void evaluate_shouldReturnAllPatientsThatHaveNeverTakenListedDrugs() throws Exception {
+    List drugSetList = new ArrayList<Concept>();
+    drugSetList.add(new Concept(88));
+    drugSetList.add(new Concept(792));
+    cohortDefinition.setDrugSets(drugSetList);
+    cohortDefinition.setWhich("NONE");
+    Cohort cohort = Context.getService(CohortDefinitionService.class).evaluate(cohortDefinition, null);
+    System.out.println("Taking Drugs None" + cohort.getMemberIds() +  " " + cohortDefinition + "===");
+    Assert.assertTrue(cohort.contains(21));
+    Assert.assertTrue(cohort.contains(22));
+    Assert.assertEquals(2, cohort.size());
+  }
+  
+  @Test
+  public void evaluate_shouldReturnAllPatientsThatHaveTakenAllListedDrugs() throws Exception {
+    List drugSetList = new ArrayList<Concept>();
+    drugSetList.add(new Concept(88));
+    drugSetList.add(new Concept(792));
+    cohortDefinition.setDrugSets(drugSetList);
+    cohortDefinition.setWhich("ALL");
+    Cohort cohort = Context.getService(CohortDefinitionService.class).evaluate(cohortDefinition, null);
+    System.out.println("Taking Drugs ALL" + cohort.getMemberIds() + " " + cohortDefinition);
+    Assert.assertTrue(cohort.contains(2));
+    Assert.assertEquals(1, cohort.size());
+
+  }
+
+  // @Test
+  // public void evaluate_shouldReturnAllPatientsTakingDrugsContainingIngredients() throws Exception {
+
+  // }
+  
+  @Test
+  public void evaluate_shouldReturnAllPatientsTakingDrugsBeforeSpecifiedDate() throws Exception {
+    cohortDefinition.setActivatedOnOrBefore(DateUtil.getDateTime(2008, 8, 2));
+    Cohort cohort = Context.getService(CohortDefinitionService.class).evaluate(cohortDefinition, null);
+    System.out.println("Taking Drugs in before " + cohort.getMemberIds() + " " + cohortDefinition);
+    Assert.assertTrue(cohort.contains(2));
+    Assert.assertTrue(cohort.contains(8));
+    Assert.assertTrue(cohort.contains(22));
+    Assert.assertEquals(3, cohort.size());
+  }
+  
+  @Test
+  public void evaluate_shouldReturnAllPatientsTakingDrugsAfterSpecifiedDate() throws Exception {
+    cohortDefinition.setActivatedOnOrAfter(DateUtil.getDateTime(2008, 8, 10));
+    Cohort cohort = Context.getService(CohortDefinitionService.class).evaluate(cohortDefinition, null);
+    System.out.println("Taking Drugs before " + cohort.getMemberIds() + " " + cohortDefinition);
+    Assert.assertTrue(cohort.contains(2));
+    Assert.assertEquals(1, cohort.size());
+
+  }
+  
+  @Test
+  public void evaluate_shouldReturnAllPatientsTakingAnyDrugWithinADateRange() throws Exception {
+    cohortDefinition.setActivatedOnOrAfter(DateUtil.getDateTime(2008, 8, 1));
+    cohortDefinition.setActivatedOnOrBefore(DateUtil.getDateTime(2008, 8, 8));
+    Cohort cohort = Context.getService(CohortDefinitionService.class).evaluate(cohortDefinition, null);
+    System.out.println("Taking Drugs in date range " + cohort.getMemberIds() + " " + cohortDefinition);
```

Comment:
Did you take a look at this? https://wiki.openmrs.org/display/docs/Java+Conventions#JavaConventions-Donotprintout